src/components/Store.react.js

```diff
@@ -2,27 +2,36 @@ import R from 'ramda';
 import React from 'react';
 import PropTypes from 'prop-types';
 
-function dataCheck(data, old) {
+/**
+ * Deep equality check to know if the data has changed.
+ *
+ * @param {any} newData - New data to compare
+ * @param {any} oldData - The old data to compare
+ * @returns {boolean} The data has changed.
+ */
+function dataChanged(newData, oldData) {
     // Assuming data and old are of the same type.
-    const oldNull = R.isNil(old);
-    const newNull = R.isNil(data);
+    const oldNull = R.isNil(oldData);
+    const newNull = R.isNil(newData);
```

Comment:
`isNil` catches both `undefined` and `null` - and mostly that distinction is moot because `undefined` can't be sent via JSON, I guess. But we should be careful about it in objects, like `{a: null} <-> {}`